awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -567,6 +574,30 @@ def create_if_needed(self, existing_item, new_item, endpoint, on_create=None, it
         else:
             self.exit_json(**self.json_output)
 
+    # We need to be able to recursevly step through fields in the case of inputs for credentials.
+    # They are dicts and we can't just compare them at the top level because the dict from the tower api may have more fields that we have.
+    # For example, say someone did:
+    #  - tower_credential:
+    #      name: 'a cred'
+    #      username: 'John'
+    # Our new dict would be like { 'username': 'new_name' }
+    # But the existing cred from tower might come back as: { 'username': 'new_name', 'password': '$encrypted$', 'field2': 'something else' }
+    @staticmethod
+    def compare_fields(new_item, existing_item):
+        needs_update = False
+        for field in new_item:
```

Comment:
To clarify my wall-of-text above, I think this would work as

```python
for field in set(new_item.keys()) | set(existing_item.keys()):
```

Sorry, I didn't get that on my first pass on this review.

The method still makes sense to me for the purpose of treating various false-y values as equal, which we discussed before.